File: rd_core/models/enums.py
"""Перечисления для моделей данных"""
from enum import Enum


class BlockType(Enum):
    """Типы блоков разметки (3 типа: текст, картинка, штамп)"""

    TEXT = "text"
    IMAGE = "image"
    STAMP = "stamp"


class BlockSource(Enum):
    """Источник создания блока"""

    USER = "user"  # Создан пользователем вручную
    AUTO = "auto"  # Создан автоматической сегментацией


class ShapeType(Enum):
    """Тип формы блока"""

    RECTANGLE = "rectangle"  # Прямоугольник
    POLYGON = "polygon"  # Многоугольник

File: web/src/models/enums.ts
export const BlockType = {
  TEXT: "text",
  IMAGE: "image",
  STAMP: "stamp",
} as const;
export type BlockType = (typeof BlockType)[keyof typeof BlockType];

export const BlockSource = {
  USER: "user",
  AUTO: "auto",
} as const;
export type BlockSource = (typeof BlockSource)[keyof typeof BlockSource];

export const ShapeType = {
  RECTANGLE: "rectangle",
  POLYGON: "polygon",
} as const;
export type ShapeType = (typeof ShapeType)[keyof typeof ShapeType];

export const ViewerState = {
  IDLE: "idle",
  DRAWING_RECT: "drawing_rect",
  DRAWING_POLYGON: "drawing_polygon",
  SELECTING: "selecting",
  MOVING_BLOCK: "moving_block",
  RESIZING_BLOCK: "resizing_block",
  PANNING: "panning",
} as const;
export type ViewerState = (typeof ViewerState)[keyof typeof ViewerState];
